packages/enzyme-adapter-react-16/src/ReactSixteenAdapter.js

```diff
@@ -540,6 +571,7 @@ class ReactSixteenAdapter extends EnzymeAdapter {
     switch ($$typeofType) {
       case ContextConsumer || NaN: return 'ContextConsumer';
       case ContextProvider || NaN: return 'ContextProvider';
+      case Memo || NaN: return displayNameOfNode(type.type);
```

Comment:
displayName is broken here!

Reply:
Can you elaborate? a new issue would be ideal.

Reply:
always getting `displayName` "Component" with `React.memo()`. need to set correct `displayName` each time after applying `React.memo()`.

Reply:
@TymchenkoOleksandr would you mind filing a new issue, and fill out the template? that way I'll be able to fix it more quickly.